### PR TITLE
[Add Check]: external file existence for ImageSeries

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
First step to resolving #87 

If an ImageSeries uses `external_file` rather than `data`; this value can either be an external link or a local uri. If local, checks that the file referenced exists. A future check will also verify that it is a relative instead of absolute path. If a url, then verifies the url can be accessed.

### CODE REVIEW

File: nwbinspector/checks/image_series.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from ..register_checks import register_check, Importance, InspectorMessage
+
+
+@register_check(importance=Importance.CRITICAL, neurodata_type=ImageSeries)
+def check_image_series_external_file_valid(image_series: ImageSeries):
+    """Check if the external_file specified by an ImageSeries actually exists at the relative location."""
+    if image_series.external_file is None:
+        return
+    for file_path in image_series.external_file:
+        if not Path(file_path).exists():
```
Comment (bendichter) [inline nwbinspector/checks/image_series.py]: path should be relative wrt nwbfile location, not current working directory. Would this work if the NWB file was in nested directories?

Comment (CodyCBakerPhD) [reply nwbinspector/checks/image_series.py]: Oh, uhh, didn't think about that. Lemme work on that then

Comment (CodyCBakerPhD) [reply nwbinspector/checks/image_series.py]: OK added a tool function for determining NWBFile path on disk from only the input object.

Comment (CodyCBakerPhD) [reply nwbinspector/checks/image_series.py]: Also added tests for different file locations relative to the NWBFile location.

File: nwbinspector/checks/image_series.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@register_check(importance=Importance.CRITICAL, neurodata_type=ImageSeries)
+def check_image_series_external_file_valid(image_series: ImageSeries):
+    """Check if the external_file specified by an ImageSeries actually exists at the relative location."""
+    if image_series.external_file is None:
+        return
+    nwbfile_path = Path(get_nwbfile_path_from_internal_object(obj=image_series))
+    for file_path in image_series.external_file:
+        if not Path(file_path).is_absolute() and not (nwbfile_path / file_path).exists():
+            yield InspectorMessage(
+                message="The external file does not exist. Please confirm the relative location to the NWBFile."
```
Comment (bendichter) [inline nwbinspector/checks/image_series.py]:
```suggestion
                message=f"The external file {file_path} does not exist. Please confirm the relative location to the NWBFile."
```

File: nwbinspector/tools.py [hunk 3]
```diff
@@ -15,3 +15,13 @@ def all_of_type(nwbfile: NWBFile, neurodata_type):
...
+
+
+def get_nwbfile_path_from_internal_object(obj):
+    """Determine the file path on disk for a NWBFile given only an internal object of that file."""
+    ancestor = obj.get_ancestor()
+    if ancestor is None:
+        return
+    while ancestor.get_ancestor() is not None:
+        ancestor = ancestor.get_ancestor()
+    return ancestor.container_source
```
Comment (bendichter) [inline nwbinspector/tools.py]: I think this function could be optimized. It doesn't currently work if given an `NWBFile`.

Comment (bendichter) [inline nwbinspector/tools.py]:
```suggestion
from pynwb.file import NWBFile

def get_nwbfile_path_from_internal_object(obj):
    """Determine the file path on disk for a NWBFile given only an internal object of that file."""
    if isinstance(obj, NWBFile):
        return obj.container_source
    else:
        return obj.get_ancestor("NWBFile").container_source
```

File: nwbinspector/tools.py [hunk 4]
```diff
@@ -15,3 +15,11 @@ def all_of_type(nwbfile: NWBFile, neurodata_type):
...
         if isinstance(obj, neurodata_type):
             yield obj
+
+
+def get_nwbfile_path_from_internal_object(obj):
+    """Determine the file path on disk for a NWBFile given only an internal object of that file."""
+    if isinstance(obj, NWBFile):
+        return obj.container_source
+    else:
+        return obj.get_ancestor("NWBFile").container_source
```
Comment (bendichter) [inline nwbinspector/tools.py]:
```suggestion
    return obj.get_ancestor("NWBFile").container_source
```

**Review by bendichter — APPROVED** [review]

**Pull request merged**